doc/man3/EVP_PKEY_set_type.pod

```diff
@@ -0,0 +1,79 @@
+=pod
+
+=head1 NAME
+
+EVP_PKEY_set_type, EVP_PKEY_set_type_str, EVP_PKEY_set_type_by_keymgmt
+- functions to change the EVP_PKEY type
+
+=head1 SYNOPSIS
+
+ #include <openssl/evp.h>
+
+ int EVP_PKEY_set_type(EVP_PKEY *pkey, int type);
+ int EVP_PKEY_set_type_str(EVP_PKEY *pkey, const char *str, int len);
+ int EVP_PKEY_set_type_by_keymgmt(EVP_PKEY *pkey, EVP_KEYMGMT *keymgmt);
+
+=head1 DESCRIPTION
+
+All the functions described here behave the same in so far that they
+clear all the previous key data and methods from I<pkey>, and reset it
+to be of the type of key given by the different arguments.  If
+I<pkey> is NULL, these functions will still return the same return
+values as if it wasn't.
+
+EVP_PKEY_set_type() initialises I<pkey> to contain an internal legacy
+key.  When doing this, it finds a L<EVP_PKEY_ASN1_METHOD(3)>
+corresponding to I<type>, and associates I<pkey> with the findings.
+It is an error if no L<EVP_PKEY_ASN1_METHOD(3)> could be found for
+I<type>.
+
+EVP_PKEY_set_type_str() initialises I<pkey> to contain an internal legacy
+key. When doing this, it finds a L<EVP_PKEY_ASN1_METHOD(3)>
+corresponding to I<str> that has then length I<len>, and associates
+I<pkey> with the findings.
+It is an error if no L<EVP_PKEY_ASN1_METHOD(3)> could be found for
+I<type>.
+
+For both EVP_PKEY_set_type() and EVP_PKEY_set_type_str(), I<pkey> gets
+a numeric type, which can be retrieved with L<EVP_PKEY_id(3)>.  This
+numeric type is taken from the L<EVP_PKEY_ASN1_METHOD(3)> that was
+found, and is equal to or closely related to I<type> in the case of
+EVP_PKEY_set_type(), or related to I<str> in the case of
+EVP_PKEY_set_type_str().
+
+EVP_PKEY_set_type_by_keymgmt() initialises I<pkey> to contain an
+internal provider side key.  When doing this, it associates I<pkey>
+with I<keymgmt>.  For keys initialised like this, an attempt is made
+to find a corresponding name that EVP_PKEY_set_type_str() accepts
+(returns 1 for), and in that case, it also saves away that
+information, more about that in L</WARNINGS> below.  When initialised
+like this, the numeric type retrieved with L<EVP_PKEY_id(3)> will
+always be B<EVP_PKEY_NONE>.
+
+=head1 WARNINGS
+
+An B<EVP_PKEY> that has been initialised with EVP_PKEY_set_type_by_keymgmt()
+can at any time be downgraded to contain a legacy key instead of a
```

Comment:
Err this seems like a bad idea to present to a end user..

Reply:
What do you suggest? Just shut up about it?

Reply:
Maybe - or I should :)

Reply:
Isn't this an internal implementation detail? Why does the end user care?

Reply:
Ok, you're both saying the same thing.  That paragraph is gonners